### PR TITLE
Fix Edge browser selection for multi-signer APKs (MSAL #2414), Fixes AB#3611725

### PR DESCRIPTION
**Pull request by shahzaibj**
Fixes [AB#3611725](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/3611725)  tracks the common-side fix for the MSAL customer issue [#2414](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/2414).

## Customer-reported issue
Resolves the root cause of [microsoft-authentication-library-for-android#2414](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/2414): when Microsoft Edge is the user''s default browser, MSAL rejects it with `Browser: com.microsoft.emmx signature hash not match` and falls back to WebView. Customers using Edge as the default cannot complete sign-in via custom tabs.

## Root cause
Microsoft Edge ships its production APK with **two** signing certificates (APK Signature Scheme v3 lineage). Historically, host apps targeting < API 28 only saw the first signer, so a one-hash safelist entry was sufficient. Once host apps started targeting API 28+, `PackageManager` started returning **all** signers, and `AndroidBrowserSelector.matches()` rejected Edge because it required strict `Set.equals` between the safelist and the browser''s actual signatures. This is a semantic bug in the matcher, not a stale hash -- but the safelist is also one hash short.

### 1. `AndroidBrowserSelector.matches()` -- relax signature comparison
Replace strict `Set.equals` with `Collections.disjoint()`. A browser is trusted when **any** of its signers appears in the descriptor''s trusted set. This is:
- the standard semantic for signature trust (matches AppAuth''s `BrowserSelector` behavior),
- safe -- `PackageInfo.signatures` is what Android verified at install time and cannot be forged,
- forward-compatible with future Edge / Chrome / other multi-signer or post-rotation browsers.

### 2. `BrowserDescriptor.getBrowserDescriptorForEdge()` -- add rotated hash
Add the second Edge signing certificate hash to the switch-browser safelist so the hard-coded Edge descriptor matches both legacy and post-rotation Edge installs even before the matcher change propagates downstream.

### 3. Regression tests
- `testSelect_Browser_multiSignerBrowserMatchesWithSubsetSafelist` -- multi-signer browser + single-hash safelist must match (the bug).
- `testSelect_Browser_signatureMismatchIsRejected` -- no overlap between presented signatures and safelist must still reject (guards the security boundary).
- New `BrowserDescriptorTest` in common4j pins both Edge hashes and the Chrome entry as regression guards.

## Companion PR
A companion PR updates `msal_default_config.json` and `auth_config.template.json` in the MSAL repo: AzureAD/microsoft-authentication-library-for-android#2515 (work item 3611726).

## Test plan
- New unit tests pass (couldn''t run locally -- Maven feed requires VSTS auth not available in this session; CI will validate).
- Existing `AndroidBrowserSelectorTest` cases unchanged in behavior under the new matcher (verified by inspection -- all existing fixtures use identical signature sets between descriptor and browser, so `disjoint == false` corresponds exactly to `equals == true`).

## Risk
**Low.** The matcher change is a one-line semantic relaxation that only changes behavior when a browser''s signature set is a proper superset/overlap (not strict equality) with a safelisted descriptor -- exactly the scenario the customer hit. Strict mismatches are still rejected.

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/ui/BrowserDescriptor.java [hunk 1]
```diff
@@ -78,6 +78,10 @@ public BrowserDescriptor(
     static private BrowserDescriptor getBrowserDescriptorForEdge() {
         final HashSet<String> edgeSignatureHashes = new HashSet<>();
         edgeSignatureHashes.add("Ivy-Rk6ztai_IudfbyUrSHugzRqAtHWslFvHT0PTvLMsEKLUIgv7ZZbVxygWy_M5mOPpfjZrd3vOx3t-cA6fVQ==");
+        // Edge production APK is signed with two certificates (APK Signature Scheme v3 lineage).
+        // Both must be present in the safelist because apps targeting API 28+ see all signers,
+        // and the safelist match accepts any signer that intersects this set.
```
[the remark this answers is not in the document]
Comment (shahzaibj) [reply common4j/src/main/com/microsoft/identity/common/java/ui/BrowserDescriptor.java]: Good catch -- fixed in e54de94. Reworded the comment so it no longer implies a strict requirement; clarified that the matcher accepts a browser when at least one of its signers is in this set, and that listing both hashes covers Edge installs at any point in the rotation lifecycle. Also added a BrowserDescriptorTest unit test in common4j that pins both Edge hashes as a regression guard.

**Review by mohitc1 — APPROVED** [review]
:shipit:

**Review by rpdome — APPROVED** [review]

**Review by mohitc1 — APPROVED** [review]
